sync/rpc/server.go

```diff
@@ -1224,6 +1241,20 @@ func (s *SyncAgentServer) FileSync(ctx context.Context, req *FileSyncRequest) (*
 }
 
 func (s *SyncAgentServer) syncFilesFromHealthyReplica(fromServer string, disks []string) (err error) {
+	defer func() {
+		if err != nil {
+			s.RebuildInfo.Lock()
+			defer s.RebuildInfo.Unlock()
+			s.RebuildInfo.IsRebuilding = false
+			s.RebuildInfo.State = types.ProcessStateError
+			s.RebuildInfo.Error = err.Error()
+		}
+	}()
+	total := len(disks)
+	s.RebuildInfo = NewRebuildStatus(total)
```

Comment:
Can we init `s.RebuildInfo` in func `NewSyncAgentServer()` and calculate `progress` in the below loop? Like `SnapshotPurge()`

The implementation here is a little bit different from the implementation of `SnapshotPurge()`. You can check which one is better/simpler and decide if you need to modify the current workflow.

Reply:
I think in the rebuild scenario, It might not make sense to create a `rebuildStatus` object when the sync agent starts as this scneario may or may not happen. It would be better to create it on-demand in my opinion.